lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({
-          children: <button className="nx-dropdown-button">Foo</button>,
-          onToggleCollapse,
-          isOpen: true,
-          onCloseClick: e => e.preventDefault()
-        }, { attachTo: container });
+  it('does not call onToggleCollapse if a click happens when onCloseClick preventsDefault', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
+
+    const props: Partial<Props> = {
+      children: <button className="nx-dropdown-button" data-testid="dropdown-button">Foo</button>,
+      onToggleCollapse,
+      isOpen: true,
+      onCloseClick: e => e.preventDefault()
+    };
+
+    const { rerender } = quickRender(props);
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
+    rerender(<NxDropdown {...minimalProps} {...props} />);
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      component.find('.nx-dropdown-button').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    const dropdownButton = screen.getByTestId('dropdown-button');
+    const toggleButton = screen.getByRole('button', { name: 'dropdown-toggle'});
+
+    await user.click(dropdownButton);
+    rerender(<NxDropdown {...minimalProps} {...props} />);
+
     expect(onToggleCollapse).not.toHaveBeenCalled();
 
-    act(() => {
-      component.find('button.nx-dropdown__toggle').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(toggleButton);
+    rerender(<NxDropdown {...minimalProps} {...props} />);
+
     expect(onToggleCollapse).not.toHaveBeenCalled();
   });
 
   it('provides onCloseClick with an event object where the typical properties work correctly', function() {
     let evt: MouseEvent | undefined,
-
         // currentTarget is only set on the event object during the event handler, so to keep it around for assertions
         // we need to store it in a separate variable
         currentTarget: EventTarget | undefined;
 
     const onCloseClick = (event: MouseEvent) => {
-          evt = event;
-          currentTarget = evt.currentTarget || undefined;
-        },
-        component = getMountedComponent({ isOpen: true, onCloseClick }, { attachTo: container });
+      evt = event;
+      currentTarget = evt.currentTarget || undefined;
+    };
+
+    const component = renderEl({ isOpen: true, onCloseClick })!;
 
     expect(evt).toBeUndefined();
 
-    act(() => {
-      component.getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
+    fireEvent.click(component);
```

Comment:
Why not `userEvent`?

Reply:
> Why not `userEvent`?

Updated here: c79a3f815f7ed7b13d789a32448d3b5332fe1974